### ISSUE TITLE
Passing `--config` with file path causes java.lang.ClassCastException

### ISSUE THREAD

**Issue by ethpran** [borkdude/clj-kondo#719]
**version**

clj-kondo v2019.12.14

**platform**

Native

**editor**

Intellij. Cursive

**problem**

passing a file path of my config file causes  java.lang.ClassCastException

**repro**

`clj-kondo --config some/path/config.edn --lint ./src`

```
Unexpected error. Please report an issue.
java.lang.ClassCastException: java.lang.String cannot be cast to java.io.File
	at clj_kondo.impl.core$read_edn_file$fn__6112.invoke(core.clj:51)
	at clj_kondo.impl.core$read_edn_file.invokeStatic(core.clj:51)
	at clj_kondo.impl.core$resolve_config.invokeStatic(core.clj:62)
	at clj_kondo.core$run_BANG_.invokeStatic(core.clj:84)
	at clj_kondo.main$main.invokeStatic(main.clj:85)
	at clj_kondo.main$main.doInvoke(main.clj:78)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at clj_kondo.main$_main$fn__6258.invoke(main.clj:104)
	at clj_kondo.main$_main.invokeStatic(main.clj:104)
	at clj_kondo.main$_main.doInvoke(main.clj:102)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clj_kondo.main.main(Unknown Source)
```

**expected behavior**

Uses the config in my file as described [here](https://github.com/borkdude/clj-kondo/blob/master/doc/config.md)

I will submit a pull request. it just seems that `clj-kondo.impl.core/read-edn-file` function expects a java.io.File, but the cli arg gets passed as string

**Comment by borkdude** [comment]
@ethpran Thanks for the fix. I'll leave this open so I can remind myself to write a unit test for it as well, so it will break if we change something.

**Comment by ethpran** [comment]
what exactly do you want to test? if you give me some direction, i dont mind writing something. im just not 100% familiar with the project so i want to make sure i hit the right points :)

**Comment by borkdude** [comment]
@ethpran In clj_kondo/main_test.clj the executable is tested using `(lint! "(foobar)" ...)` where on the dots you provide the command line arguments. There we could add a test for loading a config from a file. PR welcome.

**Comment by ethpran** [comment]
@borkdude the test looks like its going to be a bit more involved, so i'd like to ask for some guidance/opinions. `lint-jvm` function [here](https://github.com/borkdude/clj-kondo/blob/master/test/clj_kondo/test_utils.clj#L78) only knows to interpret IPersistentMap implementation as config. Otherwise it sets config passed here to nil and no longer does anything with it. In my test I'd like to pass java.io.File which obviously doesn't meet this condition. So it seems like I would need to update this condition to allow my config to pass. But even if I did this, I don't know how to treat `base-config`, since this always gets merged with the arg passed and I want to preserve the config being in File type. I could slurp my file, do the merge, and override the file contents, but I wanted to run it by you before going through with this approach. WDYT?

**Comment by borkdude** [comment]
@ethpran OK, an easy way to still get this tested but not use the `lint-jvm` stuff is just test it from `clj-kondo.core` (which is tested in `clj-kondo.core-test`).

**Comment by ethpran** [comment]
alright. ill have a look
